### PR TITLE
SpriteFrameCache load from plist file content data

### PR DESCRIPTION
**Pull request by rny**
add

SpriteFrameCache::addSpriteFramesWithFileContent()
SpriteFrameCache::removeSpriteFramesFromFileContent()

this is a fix for #7566

how to add test case?

### CODE REVIEW

**Comment by minggo** [comment]
You can add a test case in cpp-tests/Classes/SpriteTest/SpriteTest.cpp.

**Comment by rny** [comment]
test case added

**Comment by minggo** [comment]
[ci rebuild]

**Comment by minggo** [comment]
[ci  rebuild]

**Comment by minggo** [comment]
[ci rebuild]

File: cocos/2d/CCSpriteFrameCache.cpp [hunk 1]
```diff
@@ -210,6 +210,12 @@ void SpriteFrameCache::addSpriteFramesWithFile(const std::string& plist, Texture
     _loadedFileNames->insert(plist);
 }
 
+void SpriteFrameCache::addSpriteFramesWithFileContent(const std::string& plist_content, Texture2D *texture)
```
Comment (minggo) [inline cocos/2d/CCSpriteFrameCache.cpp:213]: Would you please change `plist_content` to `plistContent` to obey cocos2d-x coding style?
Thanks.

**Pull request merged**